internal/engine/evaluator.go

```diff
@@ -344,6 +365,14 @@ func (rpe *resourcePolicyEvaluator) Evaluate(ctx context.Context, tctx tracer.Co
 			if err != nil {
 				return nil, err
 			}
+
+			if p.ScopePermissions == policyv1.ScopePermissions_SCOPE_PERMISSIONS_REQUIRE_PARENTAL_CONSENT_FOR_ALLOWS {
+				for _, a := range result.unresolvedActions() {
+					if _, ok := requireParentalConsent[a]; !ok {
+						result.setEffect(a, EffectInfo{Effect: effectv1.Effect_EFFECT_DENY, Policy: noMatchScopePermissions, Scope: input.Resource.Scope})
```

Comment:
This is tricky 🤔 
If we want to defer to the parents for the final decision, should we deny the unresolved actions here? Shouldn't all those actions that didn't get an explicit DENY be passed on up the chain? One of the parent policies could have a rule that matches the unresolved action.

Reply:
Hmm. My interpretation of the desired behaviour was that this `scopePermission` mode requires an explicit match/`ALLOW` in order to pass up for further evaluation 🤔 

I think this differs from the weird role policy case because resource policies are mapped 1:1 with the request entities (unlike roles, which can return more than one role policy and should be evaluated independently).

Is there a particular use case that I'm missing here, or would this come down to a "we think it should behave like A it B"?

Reply:
I think that one of the common use cases for `REQUIRE_PARENTAL_CONSENT` mode would be to define all your access rules at the top scope and then using the child scopes to explicitly deny certain actions. In that case, the child scopes won't have rules defined for most of the available actions and they would fall into the `unresolved` group here. Otherwise, the users would have to copy all the rules from the parent down to every child -- which kind of defeats the purpose.

You could argue that the above use case should be done with `OVERRIDE_PARENT`. The issue is that then it would allow a child to (intentionally or unintentionally) allow something prohibited by the parent as well.

Reply:
> I think that one of the common use cases for `REQUIRE_PARENTAL_CONSENT` mode would be to define all your access rules at the top scope and then using the child scopes to explicitly deny certain actions.

I feel like this opens up the opportunity to write riskier permissions as it would require a policy author to remember to add every explicit `DENY`. The `DENY_ALL` with explicit `ALLOW`s approach, although potentially more laborious, removes that risk. I also like that it would be consistent with role policy behaviour.

That said, like with your example above, perhaps our users would prefer less handholding and a more efficient authoring flow. I'm very much torn.

Reply:
I see your point about being consistent with role policies. But, at the same time, resource policies are more complicated because the rules can have conditions and outputs and so on. Requiring users to duplicate those rules at every scope level is not great DX. It's quite laborious and leaves room for errors and mistakes. Therefore, I think that we shouldn't expect the child resource policies to cover all possible actions. 

In terms of implementation, I think the tricky part is keeping track of the approval chain. Perhaps we could change `unresolved` to `map[string]int`? 

For each policy in policy set:
- Go through every action in `unresolved` and set its counter to 0
- If there's a DENY rule for the action, it's denied. Full stop.
- If there's an ALLOW rule for the action, put it back into the `unresolved` pile with a counter value of 1.
- If there's no rule matching the action, put it back into the `unresolved` pile with a counter value of 0. 

At the end, go through `unresolved`:
- Any action with a counter of 1 is ALLOWED
- Everything else is DENIED

Reply:
So to clarify, the behaviour you're proposing is for resource policies with `REQUIRE_PARENTAL_CONSENT` to do the following:

- Explicit `EFFECT_DENY` -> `DENY` immediately
- Explicit `EFFECT_ALLOW` -> pass up to parent scopes for further evaluation
- No match -> pass up to parent scopes for further evaluation

So for resource policies, the only behavioural difference between `OVERRIDE_PARENT` and `REQUIRE_PARENTAL_CONSENT` is for matching rules with `EFFECT_ALLOW`:

- `OVERRIDE_PARENT`: immediate `EFFECT_ALLOW`
- `REQUIRE_PARENTAL_CONSENT`: acts as though it's an `EFFECT_NO_MATCH`

Does this sound right?

Reply:
Yes, that sounds right.